deepchem/models/tensorgraph/sequential.py

```diff
@@ -0,0 +1,114 @@
+  verbose construction of simple deep learning model.
+
+  Example
+  -------
+
+  >>> import deepchem as dc
+  >>> import numpy as np
+  >>> from deepchem.models.tensorgraph import layers
+  >>> # Define Data
+  >>> X = np.random.rand(20, 2)                     
+  >>> y = [[0, 1] for x in range(20)]
+  >>> dataset = dc.data.NumpyDataset(X, y)                              
+  >>> model = dc.models.Sequential(learning_rate=0.01)                  
+  >>> model.add(layers.Dense(out_channels=2))                                  
+  >>> model.add(layers.SoftMax())
+  """
+
+  def __init__(self, **kwargs):
+    """Initializes a sequential model
+    """
+    self.num_layers = 0
+    self._prev_layer = None
+    if "use_queue" in kwargs:
+      if kwargs["use_queue"]:
+        raise ValueError("Sequential doesn't support queues.")
+    kwargs["use_queue"] = False
+    self._layer_list = []
+    self._built = False
+    super(Sequential, self).__init__(**kwargs)
+
+  def add(self, layer):
+    """Adds a new layer to model.
+
+    Parameter
+    ---------
+    layer: Layer
+      Adds layer to this graph.
+    """
+    self._layer_list.append(layer)
+
+  def fit(self, dataset, loss, **kwargs):
+    """Fits on the specified dataset.
+
+    If called for the first time, constructs the TensorFlow graph for this
+    model. Fits this graph on the specified dataset according to the specified
+    loss.
+
+    Parameters
+    ----------
+    dataset: dc.data.Dataset
+      Dataset with data
+    loss: string
+      Only "binary_crossentropy" or "mse" for now.
+    """
+    X_shape, y_shape, _, _ = dataset.get_shape()
+    # Calling fit() for first time
+    if not self._built:
+      feature_shape = X_shape[1:]
+      label_shape = y_shape[1:]
+      # Add in features
+      features = Feature(shape=(None,) + feature_shape)
+      self._add_layer(features)
+      # Add in labels
+      labels = Label(shape=(None,) + label_shape)
+      self._add_layer(labels)
+
+      # Add in all layers
+      prev_layer = features
+      if len(self._layer_list) == 0:
+        raise ValueError("No layers have been added to model.")
+      for ind, layer in enumerate(self._layer_list):
+        if not len(layer.in_layers) == 0:
+          raise ValueError("Cannot specify in_layers for Sequential.")
+        layer.in_layers += [prev_layer]
+        self._add_layer(layer)
+        prev_layer = layer
+      # The last layer is the output of the model
+      self.outputs.append(prev_layer)
+
+      if loss == "binary_crossentropy":
+        smce = SoftMaxCrossEntropy(in_layers=[labels, prev_layer])
+        self._add_layer(smce)
+        self.set_loss(ReduceMean(in_layers=[smce]))
+      elif loss == "mse":
+        mse = ReduceSquareDifference(in_layers=[prev_layer, labels])
+        self._add_layer(mse)
+        self.loss = mse
+      else:
+        # TODO(rbharath): Add in support for additional losses.
+        raise ValueError("Unsupported loss.")
+    self._built = True
```

Comment:
Why is this needed?  super().fit() should call build and will also install the queue for faster training/

Reply:
Good point. Will remove.